state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1495,12 +1507,21 @@ pub mod pallet {
 				);
 
 				let mut stable_amount = swap.stable_amount.unwrap_or_default();
+				swap.stable_amount_before_fees = Some(stable_amount);
 
 				for fee_type in &swap.swap.fees {
 					let remaining_amount = match fee_type {
 						FeeType::NetworkFee { min_fee_enforced } => {
-							let FeeTaken { remaining_amount, fee } =
-								Self::take_network_fee(stable_amount, *min_fee_enforced);
+							let FeeTaken { remaining_amount, fee } = Self::take_network_fee(
+								stable_amount,
+								if *min_fee_enforced {
+									MinFeePolicy::Enforced {
+										swap_request_id: swap.swap.swap_request_id,
+									}
+								} else {
+									MinFeePolicy::NotEnforced
+								},
+							);
```

Comment:
not related to this PR specifically, but there doesn't seem to be a need for the `let mut stable_amount` variable. We only assign to it to later assign to swap... we could just assign to swap directly?